libraries/AP_Camera/AP_Camera.cpp

```diff
@@ -271,42 +268,44 @@ void AP_Camera::send_feedback(mavlink_channel_t chan, AP_GPS &gps, const AP_AHRS
 }
 
 
-/*  update location, for triggering by GPS distance moved
-    This function returns true if a picture should be taken
-    The caller is responsible for taking the picture based on the return value of this function.
-    The caller is also responsible for logging the details about the photo
+/*  update; triggers by distance moved
 */
-bool AP_Camera::update_location(const struct Location &loc, const AP_AHRS &ahrs)
+void AP_Camera::update()
 {
+    if (gps.status() < AP_GPS::GPS_OK_FIX_3D) {
```

Comment:
I'll say that I'm not particularly happy about this assumption. You might have lost GPS fix but still have a good position estimate if you have other sensors, like optical flow.